aptos-move/framework/src/natives/cryptography/ed25519.rs

```diff
@@ -136,17 +138,59 @@ pub struct GasParameters {
 }
 
 pub fn make_all(gas_params: GasParameters) -> impl Iterator<Item = (String, NativeFunction)> {
-    let natives = [
-        // Ed25519
-        (
-            "public_key_validate_internal",
-            make_native_from_func(gas_params.clone(), native_public_key_validate),
-        ),
-        (
-            "signature_verify_strict_internal",
-            make_native_from_func(gas_params, native_signature_verify_strict),
-        ),
-    ];
-
+    let mut natives = vec![];
```

Comment:
You could've kept the `Vec` from above, no? It's cleaner/easier to red.

Then, you could similarly declare a `test_natives` `Vec` and use [`append`](https://doc.rust-lang.org/std/vec/struct.Vec.html#method.append) to append it to `natives` based on the feature flag.